src/BizHawk.Emulation.Cores/Consoles/Nintendo/QuickNES/QuickNES.cs

```diff
@@ -47,6 +48,10 @@ public QuickNES(byte[] file, QuickNESSettings settings, QuickNESSyncSettings syn
 				InitAudio();
 				InitMemoryDomains();
 
+				// Setting input callback
+				_inputCallback = InputCallback;
+				QN.qn_set_input_callback(_inputCallback);

```

Comment:
Since this is never used to unsubscribe, do you need the field? Does `() => InputCallbacks.Call()` work?

Reply:
You still need the field, to avoid GC eating the delegate. `_inputCallback = () => InputCallbacks.Call();` probably would work, yes (or even `_inputCallback = InputCallbacks.Call`)